articles/150_roslaunch.md

```diff
@@ -428,62 +428,179 @@ The remapping design document[^static_remapping] goes into detail on how you can
+# Values from message rcl_interfaces/Log
+uint8 log_level
+# Remap rules
+# TODO(sloretz) rcl_interfaces message for remap rules?
+string[] remap_rules
+# Parameters to set
+rcl_interfaces/Parameter[] parameters
+
+# key/value arguments that are specific to a type of container process
+rcl_interfaces/Parameter[] extra_arguments
+---
+# True if the node was successfully loaded
+bool success
+# Human readable error message if success is false, else empty string
+string error_messsage
+# Name of the loaded composable node (including namespace)
+string full_node_name
+# A unique identifier for the loaded node
+uint64 unique_id
+```
+
+####### unload_node
+
+```
+# Container specific unique id of a loaded node
+uint64 unique_id
+---
+# True if the node existed and was unloaded
+bool success
+# Human readable error message if success is false, else empty string
+string error_messsage
 ```
 
-Whether or not the above results in three processes (one container and two proxy processes) or a single process, probably doesn't matter to the user or at least would be fine with the user in exchange for less overhead.
+####### list_nodes
 
-It would be more efficient because you don't need to have a node in each proxy to actually make the Service call to the load that node the user wants to run, and you don't need to maintain a bond between the container and the proxy processes.
+```
+---
+# List of full node names including namespace
+string[] full_node_names
+# corresponding unique ids (must have same length as full_node_names)
+uint64[] unique_ids
+```
 
-So I'm interested in what others think about coming up with a standardized executable which can load and configure many nodes at once from command line arguments or a config file.
-</div>
+###### Exit Code
+If the container proces is asked to shutdown due to normal [Termination], then the exit code must be 0.
+If it exits due to an error then exit code must be any other number.
 
-#### By Proxy
+##### Parallel vs Sequential Loading of Nodes
 
-The other way to implement a process which dynamically instantiates ROS nodes, is to send the request to instantiate a node asynchronously from another process.
-So the "container" process would start up and provide a service (ROS Service or otherwise) that lets external processes request a node be instantiated with a given set of configurations.
+If it is possible to load multiple nodes in parallel, then it needs to be decided how to load the nodes.
+The container process should load nodes as soon as it is asked.
+It should be up to the launch system to decide whether to load nodes in parallel or sequentially.
+If multiple nodes of the same type are to be launched, then the launch system should load the nodes sequentially so each is able to remap it's name before the next is loaded.
+If they are of different types then the launch system may choose to try to load them in parallel, where the exact order they get loaded is determined by chance or the container process.
```

Comment:
Hmm I understood the ROS2 launch system was aiming to be deterministic in its execution. @wjwwood ?

Reply:
I think that the key phrase here is "may choose to".  If you want to continue to have deterministic behavior, it probably makes the most sense to load them in sequential order (perhaps as defined by the launch file?)

I could still see a case for loading in parallel where rigorous sequential guarantees aren't needed.  It should speed up launching a large set of composed nodes.